### ISSUE TITLE
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

### ISSUE THREAD

**Issue by fengpei0114** [typescript-eslint/typescript-eslint#3819]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Hi team**, we got a warning when running `yarn run eslint`:
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.4.0

YOUR TYPESCRIPT VERSION: 4.4.2

Please only submit bug reports when using the officially supported version.
```

The version we used like: 
| package | version |
| --- | --- |
| node | 12.22.5 |
| typescript | 4.4.2 |
| @typescript-eslint/parser | 4.29.3 |
| typescript-estree | 4.29.3 | 

What should I do to fix this warning?

**Comment by fr31b3u73r** [comment]
PR fixing this should already be merged: https://github.com/typescript-eslint/typescript-eslint/commit/f81831bd279a32da6dbab0f1c061053ea43965f6

**Comment by fengpei0114** [comment]
Hi @fr31b3u73r, I noticed that this PR merged 2 days ago, and the version `4.29.3` was released 7 days ago, so that means the version `4.29.3` can not fix this warning, it will be fixed in the next version?

**Comment by bradzacher** [comment]
Isn't the message pretty self explanatory?

YOUR TYPESCRIPT VERSION: 4.4.2

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.4.0

The version of our tooling you have installed does not support TS 4.4.

---

TS 4.4 was officially released 4 days ago. Please be patient!
The next release will contain support.
